hack/build/rpm-deps.sh

```diff
@@ -46,6 +46,7 @@ nbdkit-basic-filters
 nbdkit-curl-plugin
 nbdkit-xz-filter
 nbdkit-gzip-filter
+nbdkit-tar-filter
```

Comment:
We need to understand if/how nbdkit does caching with this filter.  I bet some caching is going on as it would be pretty expensive to always open/seek to the appropriate offset in the archive.  If there is any significant on disk/memory caching I'd prefer we use simply the golang tar reader to seek to the file and download it to scratch.  Then convert from scratch to target.  Thus not getting nbdkit involved at all.

Reply:
Sorry, I can't get the point, why nbdkit cache will be the concern.
I believe nbdkit-tar-filter will only retrieve tar header block, if find the header name match "tar-entry", then get the target file data offset. (I think golang tar reader will do the same thing)

Reply:
Looks to me like tar writes to tmp which is something we have to avoid:  https://github.com/libguestfs/nbdkit/blob/45b72f5bd8fc1b475fa130d06c86cd877bf595d5/filters/tar/tar.c#L146

But I'm not an expert on nbdkit.

@rwmjones care to comment on how curl + tar filter uses tmp space?

Reply:
tar filter will not use tmp space (or more accurately, only a few bytes are used for a temporary file used to parse offsets printed by tar).  Here's how it works:

https://gitlab.com/nbdkit/nbdkit/-/blob/462eb6cbf605c2969f6e851bb9076f8e3961cccb/filters/tar/tar.c#L138